Contents

knalin55 opened the pull request.

knalin55 committed:
Update README.md

knalin55 committed:
Create test

knalin55 committed:
Delete test

knalin55 committed:
Create readme.md

knalin55 committed:
Update readme.md

knalin55 committed:
Update readme.md

knalin55 committed:
Create readme.md

knalin55 committed:
Create readme.md

knalin55 committed:
Update README.md

knalin55 committed:
Add files via upload

knalin55 committed:
Update readme.md

knalin55 committed:
Update readme.md

knalin55 committed:
Add files via upload

knalin55 committed:
Rename Predictions/bt5+wmt_bilingual+wpc_en.txt to predictions/bt5+wmt_bilingual+wpc_en.txt

knalin55 committed:
Update readme.md

knalin55 committed:
Add files via upload

knalin55 committed:
Create results.md

knalin55 committed:
Update results.md

jirka-x1 commented on data/make_parquet.py:
```diff
@@ -0,0 +1,31 @@
+
+# Write first args.num_lines to a csv file. 
+    count =0
+    with open('{}.csv'.format(args.language), "w") as file:
```
The f-strings are now typically preferred over using `format`.
```suggestion
    with open(f'{args.language}.csv', "w") as file:
```

jirka-x1 commented on data/make_parquet.py:
```diff
@@ -0,0 +1,31 @@
+            file.write("SNo\tinput\n")
+            for line in r_file:
+                if line.strip("\n").strip() != "":
+                    file.write(str(count) + "\t" + line.strip("\n") + " \n")
```
Either make sure `line` does not contains `\t` or (preferably) use `csvreader` (with \t as the separator)

jirka-x1 commented on data/make_parquet.py:
```diff
@@ -0,0 +1,31 @@
+def main(args: argparse.Namespace) -> None:
+
+
+# Write first args.num_lines to a csv file. 
```
Is it csv or tsv? And you use txt ending.

jirka-x1 commented on evaluate.py:
```diff
@@ -0,0 +1,85 @@
+parser.add_argument("--finetune", default=None, type=str, help="Number of lines to keep in dataset")
+parser.add_argument("--ckpt", default=None, type=str, help="Number of lines to keep in dataset")
+
+def main(args: argparse.Namespace) -> None:
```
split into well defined functions (here and all other places)

jirka-x1 commented on evaluate.py:
```diff
@@ -0,0 +1,85 @@
+            label = ""
+            inputs = [sent.lex for i, sent in enumerate(entry.lexs) if i%2==1] if lang == "ru" else [sent.lex for i, sent in enumerate(entry.lexs)]
+            for triple in entry.list_triples():
+                
```
In several places, you have these random empty lines. Follow PEP8. Doesn't your IDE complain?

knalin55 replied:
My MS VS Code had a different linter selected as default. So it didn't show the coding style-related issues. I have fixed the scripts now. It should be better than before.

jirka-x1 commented on evaluate.py:
```diff
@@ -0,0 +1,85 @@
+        b.fill_benchmark(files)
+            
+        entry_list = [entry for entry in b.entries]
+        triples_list    = []
```
You do not explain what are triples.

jirka-x1 commented on evaluate.py:
```diff
@@ -0,0 +1,85 @@
+            for triple in entry.list_triples():
+                
+                
+                label = label + sub + " ".join(triple.split("|")[0].strip().split("_")) \
```
this is too hard to read; add a comment.
Also you could create a (commented) function for doing
```
def some_intuitive_name(text: str, some_intuitive_name_for_triple_index: int) -> str:
    return text.split("|")[some_intuitive_name_for_triple_index].strip().split("_")
```

jirka-x1 commented on pretrain.py:
```diff
@@ -0,0 +1,184 @@
+        ru = []
+        for line in dataset:
+            en.append(line.split("\t")[0])
+            ru.append(line.split("\t")[1])
```
why not:
```suggestion
        en = [line.split("\t")[0] for line in dataset]
        ru = [line.split("\t")[1] for line in dataset]
```
But again, you should create a function for getting n-th column from a dataset.

jirka-x1 commented on finetune.py:
```diff
@@ -0,0 +1,230 @@
+
+if __name__ == "__main__":
+    args = parser.parse_args([] if "__file__" not in globals() else None)
+    main(args)
```
Make sure all files contain `\n` as the last character (See 4.6 https://ufal.mff.cuni.cz/courses/npfl096-2014-su/project)

jirka-x1 commented on pretrain.py:
```diff
@@ -0,0 +1,184 @@
+parser.add_argument("--ckpt_path", default=None, type=str, help="Checkpoint Path to resume from")
+
+
+def main(args: argparse.Namespace) -> None:
```
split into functions

knalin55 committed:
Update data/make_parquet.py

Co-authored-by: jirka-x1 <jirka.hana@gmail.com>

knalin55 committed:
Update pretrain.py

https://github.com/knalin55/NPFL128-LTIP/pull/1#discussion_r967874751

knalin55 committed:
Update evaluate.py

https://github.com/knalin55/NPFL128-LTIP/pull/1/files#r967862340

knalin55 committed:
Update make_parquet.py

https://github.com/knalin55/NPFL128-LTIP/pull/1#discussion_r967862340

Added parser and the arguments in the function get_parser()

knalin55 committed:
Update finetune.py

https://github.com/knalin55/NPFL128-LTIP/pull/1#discussion_r967862340 (Added parser and arguments in get_parser())

knalin55 committed:
Update evaluate.py

Updated variable names; Added comments

knalin55 committed:
Update evaluate.py

https://github.com/knalin55/NPFL128-LTIP/pull/1#discussion_r967863012 (Resolved the issue as suggested; Added preprocessing part to a function; Added comments)

knalin55 committed:
Update pretrain.py

Co-authored-by: jirka-x1 <jirka.hana@gmail.com>

knalin55 committed:
Update make_parquet.py

updated file extension; added comments

knalin55 committed:
Update make_parquet.py

https://github.com/knalin55/NPFL128-LTIP/pull/1#discussion_r967862058 (updated code for generating (en/ru).tsv)

knalin55 committed:
Update make_parquet.py

Fixed indentations and whitespaces

knalin55 committed:
Update evaluate.py

Fixed indentations and whitespaces

knalin55 committed:
Update finetune.py

Fixed indentation and whitespaces

knalin55 committed:
Update pretrain.py

Fixed indentation and whitespaces

jirka-x1 commented on data/make_parquet.py:
```diff
@@ -1,31 +1,41 @@
+    count = 0
+    with open(f'{args.language}.tsv', "w") as file:
+    # write n_lines to (en/ru).tsv (tmp)
         with open(os.path.join(args.directory, '{}.txt'.format(args.language))) as r_file:
```
You changed the formatting to "f" string above (for the output file), but not here.

jirka-x1 commented on data/make_parquet.py:
```diff
@@ -1,31 +1,41 @@
-    df = pd.read_csv('{}.csv'.format(args.language), sep="\t")
-    df.to_parquet('{}.parquet'.format(args.language)) # Convert csv to parquet
-    os.remove("{}.csv".format(args.language)) # Remove intermediate file
+    df = pd.read_csv('{}.tsv'.format(args.language), sep="\t")
```
ditto (and elsewhere)

jirka-x1 commented on evaluate.py:
```diff
@@ -0,0 +1,104 @@
+
+    prefix = dict(en="Generate in English ", ru="Generate in Russian ")
+
+    def extract_component(triple: str, entity_index: int) -> str:
```
why is this a function under main? (the disadvantage is that you can access any variables in main (langs, args, prefix). 
Make it a normal function (and similarly the other ones)

jirka-x1 commented on evaluate.py:
```diff
@@ -0,0 +1,104 @@
+        return [re.findall(r"[a-z]*", str)[0]] +\
+            re.findall(r'[A-Z](?:[a-z]+|[A-Z]*(?=[A-Z]|$))', str)
+
+    def preprocess_data(lang):
```
what is the type of lang? All basic function arguments should have a type hint (exceptions are complex objects from third party untyped libraries, where it is not easy to find out what the type is)

jirka-x1 commented on pretrain.py:
```diff
@@ -0,0 +1,182 @@
+        # BT5_WMT/MT5_WMT: Training T5/MT5 further on WMT dataset.
+        # load and process tsv file for WMT News dataset
+
+        dataset = open("./data/news-commentary-v14.en-ru.tsv").read().split("\n")
```
yes, but obviously using a CsvReader (with \t as separator) is recommended.
